File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/LessonAllocation.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class LessonAllocation: Base
    {
        public int LessonsPerWeek { get; set; }
        public string? Description { get; set; }
        public int SubjectId { get; set; }
        public Subject? Subject { get; set; } = null;
        public int LearningLevelId { get; set; }
        public LearningLevel? LearningLevel { get; set; } = null;        
    }
}

File: SchoolWebApp-Client/src/app/cbe/assessments/models/lesson-allocation.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class LessonAllocation extends ResourceModel<LessonAllocation> {
    public lessonsPerWeek?: number;
    public description?: string;
    public subjectId?: number;
    public subject?: any;
    public learningLevelId?: number;
    public learningLevel?: any;

    // Used by add/edit form for cascading filters
    public academicYearId?: number;
    public curriculumId?: number;

    constructor(model?: Partial<LessonAllocation>) {
        super(model);
    }
}
